runtime/include/priority_queue.h

```diff
@@ -332,7 +331,7 @@ static inline int
 priority_queue_length_nolock(struct priority_queue *priority_queue)
 {
 	assert(priority_queue != NULL);
-	assert(!listener_thread_is_running());
+	if (scheduler != SCHEDULER_MTDS) assert(!listener_thread_is_running());
```

Comment:
By moving this out of the assert, you're losing the branch prediction hint.  Maybe should be:

```c
assert(scheduler == SCHEDULER_MTDS || !listener_thread_is_running());
```